packages/extension/src/lib/extensionScripts.ts

```diff
@@ -41,31 +35,20 @@ export const getContentScriptPermissionAndRegister =
 
 const contentScriptKey = 'permission_key';
 
-interface UseExtensionPermissionProps {
-  origin: string;
-}
-
-export const useExtensionPermission = ({
+export const requestContentScripts: RequestContentScripts = (
   origin,
-}: UseExtensionPermissionProps): UseExtensionPermission => {
-  const client = useQueryClient();
+  onPermission,
+) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
   const { trackEvent } = useContext(AnalyticsContext);
-  const { data: contentScriptGranted, isFetched } = useQuery(
-    contentScriptKey,
-    getContentScriptPermission,
-    {
-      refetchOnMount: false,
-      refetchOnReconnect: false,
-      refetchOnWindowFocus: false,
-      refetchIntervalInBackground: false,
-    },
-  );
+  const client = useQueryClient();
```

Comment:
Watch out this is a hook call inside a regular function. I am not sure how this even works because there is no context available. Probably preact magic but in React this would never work.

Reply:
I think those should be passed as arguments to `requestContentScripts` or similar.

Reply:
great catch, thanks 🙏